packages/playground/data-liberation/src/import/WP_Stream_Importer.php

```diff
@@ -202,19 +207,34 @@ public function set_frontloading_retries_iterator( $frontloading_retries_iterato
 	 */
 	private $importer;
 
-	public function next_step() {
+	/**
+	 * Calculate next steps in the import process.
+	 *
+	 * @param int $count The number of entities to process in one go.
+	 *
+	 * @return bool
+	 */
+	public function next_step( $count = 10000 ) {
 		switch ( $this->stage ) {
 			case self::STAGE_INITIAL:
 				$this->next_stage = self::STAGE_INDEX_ENTITIES;
 				return false;
 			case self::STAGE_INDEX_ENTITIES:
-				if ( true === $this->index_next_entities() ) {
+				if ( true === $this->index_next_entities( $count ) ) {
 					return true;
 				}
+
 				$this->next_stage = self::STAGE_TOPOLOGICAL_SORT;
 				return false;
 			case self::STAGE_TOPOLOGICAL_SORT:
-				// @TODO: Topologically sort the entities.
+				if ( true === $this->topological_sort_next_entity( $count ) ) {
+					return true;
+				}
+
+				// We indexed all the entities. Now sort them topologically.
+				$this->topological_sorter->sort_topologically();
+				$this->topological_sorter = null;
```

Comment:
It reads confusing to me:

1. Sort every entity we encounter and short circuit while we have entities
2. Once we run out of entities to sort, sort them

What would be another way to call these operations?